File: Easy/0507-Perfect Number/Solution.rs
impl Solution {
    pub fn check_perfect_number(num: i32) -> bool {
        if num <= 1 {
            return false;
        }

        let mut sum = 1;
        let mut i = 2;

        while i * i < num {
            if num % i == 0 {
                sum += i + num / i;
            }
            i += 1;
        }

        if i * i == num {
            sum += i;
        }

        sum == num
    }
}

File: Easy/0507-Perfect Number/Solution.rb
# @param {Integer} num
# @return {Boolean}
def check_perfect_number(num)
    return false if num <= 1

    sum = 1
    i = 2

    while i * i < num
        sum += i + num / i if num % i == 0
        i += 1
    end

    sum += i if i * i == num

    return sum == num
end
